src/ruby/ext/grpc/extconf.rb

```diff
@@ -92,6 +92,7 @@
 end
 
 $LDFLAGS << ' -Wl,-wrap,memcpy' if RUBY_PLATFORM =~ /linux/
+$LDFLAGS << ' -Wl,--no-undefined'
```

Comment:
I think you want to add the same if statement as above. This is gcc / Linux only, and that is currently breaking the Ruby build for MacOS.